### ISSUE TITLE
Shutdown after Transaction rejected: output already spent

### ISSUE THREAD

**Issue by mercurytoxic** [lightningnetwork/lnd#2782]
### Background

lnd shuts down right after startup with Transaction rejected: output already spent. What would be the proper way to fix/ignore/skip/delete this issue/transaction?

### Your environment

lnd version 0.5.2-99-beta commit=6516.36cc1da8 running on Testnet
Linux  4.20.3-arch1-1-ARCH x86_64
bitcoind v0.17.1.0

### Steps to reproduce

After normal shutdown and upgrade to commit https://github.com/lightningnetwork/lnd/commit/36cc1da8ea0eec6be4559c673bc07bd0ae9be66a from commit https://github.com/lightningnetwork/lnd/commit/2bf22617d4edf571b2f0b4b46f1a2d721bef74c8 

### Actual behaviour

lnd is unable to start, with error `Transaction rejected: output already spent `

```
2019-03-15 11:39:40.267 [DBG] SWPR: Best height: 1484722                                                                                                                                                                                                                                                                     
2019-03-15 11:39:40.267 [INF] NTFN: New block epoch subscription                                                                                                                                                                                                                                                             
2019-03-15 11:39:41.318 [INF] NTFN: New confirmation subscription: txid=59562377bafd4de8fe5ecad7ae69c18a64b3119d041868552869cf5048bc7336, num_confs=1                                                                                                                                                                        
2019-03-15 11:39:41.318 [DBG] NTFN: Using height hint 1484721 retrieved from cache for txid=59562377bafd4de8fe5ecad7ae69c18a64b3119d041868552869cf5048bc7336                                                                                                                                                                 
2019-03-15 11:39:41.318 [DBG] NTFN: Dispatching historical confirmation rescan for txid=59562377bafd4de8fe5ecad7ae69c18a64b3119d041868552869cf5048bc7336                                                                                                                                                                     
2019-03-15 11:39:41.318 [INF] UTXN: Commitment outpoint 59562377bafd4de8fe5ecad7ae69c18a64b3119d041868552869cf5048bc7336:2 registered for confirmation notification.                                                                                                                                                         
2019-03-15 11:39:41.318 [INF] UTXN: (Re)-sweeping 1 heights below height=1484722                                                                                                                                                                                                                                             
2019-03-15 11:39:41.318 [DBG] UTXN: Attempting to sweep outputs at height=1352192                                                                                                                                                                                                                                            
2019-03-15 11:39:41.319 [INF] UTXN: Attempting to graduate height=1352192: num_kids=0, num_babies=1                                                                                                                                                                                                                          
2019-03-15 11:39:41.319 [INF] UTXN: Publishing CLTV-delayed HTLC output using timeout tx (txid=3c92ea7e140bedb36bad976d807dc2f33b41d5e1412a6176708b7756b3b2d06e): (*wire.MsgTx)(0xc001a7a200)({                                                                                                                              
 Version: (int32) 2,                                                                                                                                                                                                                                                                                                         
 TxIn: ([]*wire.TxIn) (len=1 cap=1) {                                                                                                                                                                                                                                                                                        
  (*wire.TxIn)(0xc0000b46c0)({                                                                                                                                                                                                                                                                                               
   PreviousOutPoint: (wire.OutPoint) 59562377bafd4de8fe5ecad7ae69c18a64b3119d041868552869cf5048bc7336:0,                                                                                                                                                                                                                     
   SignatureScript: ([]uint8) {                                                                                                                                                                                                                                                                                              
   },                                                                                                                                                                                                                                                                                                                        
   Witness: (wire.TxWitness) (len=5 cap=5) {                                                                                                                                                                                                                                                                                 
    ([]uint8) {                                                                    
    },                                                                             
    ([]uint8) (len=71 cap=71) {                                         
     00000000  30 44 02 20 54 6c 18 59  85 59 6d 43 84 3b bf 6b  |0D. Tl.Y.YmC.;.k|
     00000010  b0 ee 55 cd 0e f8 2f 25  72 c7 b9 6e 72 29 38 2b  |..U.../%r..nr)8+|
     00000020  fa 6d f1 45 02 20 54 d7  ce dd c4 1b e3 0a a4 b1  |.m.E. T.........|
     00000030  49 24 05 d1 a3 98 9b ac  dd b3 f1 19 f5 e9 25 f8  |I$............%.|
     00000040  c4 59 53 44 b1 1c 01                              |.YSD...|
    },                                 
    ([]uint8) (len=71 cap=71) {
     00000000  30 44 02 20 50 6b 7c 99  97 34 c3 5b 6e e7 7d 61  |0D. Pk|..4.[n.}a|
     00000010  05 44 38 4e 04 be e7 b4  56 47 48 91 02 b5 f2 8d  |.D8N....VGH.....|
     00000020  6a 55 a7 a7 02 20 46 5c  c0 f4 7c 17 c8 1b f1 a8  |jU... F\..|.....|
     00000030  6e df b6 6d 85 45 ef 4c  8e 0d 80 39 c5 e6 e3 5d  |n..m.E.L...9...]|
     00000040  bd e8 21 d5 38 cd 01                              |..!.8..|
    },
    ([]uint8) {
    },
    ([]uint8) (len=133 cap=133) {
     00000000  76 a9 14 59 43 52 37 86  fc 62 17 6d 84 ba 57 3c  |v..YCR7..b.m..W<|
     00000010  ff 14 52 56 57 a7 cd 87  63 ac 67 21 02 2a 4c cc  |..RVW...c.g!.*L.|
     00000020  f1 5f 8d 78 d3 91 79 5c  d7 28 6d 29 79 1a d8 8f  |._.x..y\.(m)y...|                                                       
     00000030  79 9c 49 b1 36 a3 00 2f  24 8b 37 4a af 7c 82 01  |y.I.6../$.7J.|..|
     00000040  20 87 64 75 52 7c 21 02  3d f2 bd 25 63 00 65 0e  | .duR|!.=..%c.e.|
     00000050  08 6b 17 e9 7c 0f 81 c5  01 c9 03 02 47 79 bb d9  |.k..|.......Gy..|
     00000060  6f 5b b3 e8 8d c5 80 2c  52 ae 67 a9 14 e2 57 b9  |o[.....,R.g...W.|                     
     00000070  3d a8 40 1a a9 d6 87 7f  9b 52 88 46 68 88 a4 6b  |=.@......R.Fh..k|
     00000080  1f 88 ac 68 68                                    |...hh|
    }                                       
   },          
   Sequence: (uint32) 0
  })                           
 },                                                                                
 TxOut: ([]*wire.TxOut) (len=1 cap=1) {                                            
  (*wire.TxOut)(0xc0017d2280)({                                                    
   Value: (int64) 4210,                                                            
   PkScript: ([]uint8) (len=34 cap=34) {                                  
    00000000  00 20 65 57 9d 0f bc 9d  53 63 c1 04 38 95 a8 be  |. eW....Sc..8...|
    00000010  ae 32 4a 55 54 f3 e6 75  55 de d1 f7 67 9f f5 6c  |.2JUT..uU...g..l|
    00000020  fb 9f                                             |..|               
   }                                                                               
  })                                                                               
 },                                                                                
 LockTime: (uint32) 1352192                                               
})    
    
2019-03-15 11:39:41.319 [INF] LNWL: Inserting unconfirmed transaction 3c92ea7e140bedb36bad976d807dc2f33b41d5e1412a6176708b7756b3b2d06e
2019-03-15 11:39:41.320 [DBG] NTFN: Confirmation details for txid=59562377bafd4de8fe5ecad7ae69c18a64b3119d041868552869cf5048bc7336 not found during historical dispatch, waiting to dispatch at tip                                                                                                                         
2019-03-15 11:39:41.478 [INF] LNWL: Removed invalid transaction: (*wire.MsgTx)(0xc001a7a200)({
 Version: (int32) 2,                                                               
 TxIn: ([]*wire.TxIn) (len=1 cap=1) {                                              
  (*wire.TxIn)(0xc0000b46c0)({                                                     
   PreviousOutPoint: (wire.OutPoint) 59562377bafd4de8fe5ecad7ae69c18a64b3119d041868552869cf5048bc7336:0,
   SignatureScript: ([]uint8) {                                                    
   },                                                                              
   Witness: (wire.TxWitness) (len=5 cap=5) {                                       
    ([]uint8) {                                                         
    },
    ([]uint8) (len=71 cap=71) {
     00000000  30 44 02 20 54 6c 18 59  85 59 6d 43 84 3b bf 6b  |0D. Tl.Y.YmC.;.k|
     00000010  b0 ee 55 cd 0e f8 2f 25  72 c7 b9 6e 72 29 38 2b  |..U.../%r..nr)8+|
     00000020  fa 6d f1 45 02 20 54 d7  ce dd c4 1b e3 0a a4 b1  |.m.E. T.........|
     00000030  49 24 05 d1 a3 98 9b ac  dd b3 f1 19 f5 e9 25 f8  |I$............%.|
     00000040  c4 59 53 44 b1 1c 01                              |.YSD...|
    },                 
    ([]uint8) (len=71 cap=71) {         
     00000000  30 44 02 20 50 6b 7c 99  97 34 c3 5b 6e e7 7d 61  |0D. Pk|..4.[n.}a|
     00000010  05 44 38 4e 04 be e7 b4  56 47 48 91 02 b5 f2 8d  |.D8N....VGH.....|
     00000020  6a 55 a7 a7 02 20 46 5c  c0 f4 7c 17 c8 1b f1 a8  |jU... F\..|.....|
     00000030  6e df b6 6d 85 45 ef 4c  8e 0d 80 39 c5 e6 e3 5d  |n..m.E.L...9...]|
     00000040  bd e8 21 d5 38 cd 01                              |..!.8..|
    },
    ([]uint8) {            
    },
    ([]uint8) (len=133 cap=133) {
     00000000  76 a9 14 59 43 52 37 86  fc 62 17 6d 84 ba 57 3c  |v..YCR7..b.m..W<|                                                                          
     00000010  ff 14 52 56 57 a7 cd 87  63 ac 67 21 02 2a 4c cc  |..RVW...c.g!.*L.|                                      
     00000020  f1 5f 8d 78 d3 91 79 5c  d7 28 6d 29 79 1a d8 8f  |._.x..y\.(m)y...|                   
     00000030  79 9c 49 b1 36 a3 00 2f  24 8b 37 4a af 7c 82 01  |y.I.6../$.7J.|..|
     00000040  20 87 64 75 52 7c 21 02  3d f2 bd 25 63 00 65 0e  | .duR|!.=..%c.e.|
     00000050  08 6b 17 e9 7c 0f 81 c5  01 c9 03 02 47 79 bb d9  |.k..|.......Gy..|
     00000060  6f 5b b3 e8 8d c5 80 2c  52 ae 67 a9 14 e2 57 b9  |o[.....,R.g...W.|
     00000070  3d a8 40 1a a9 d6 87 7f  9b 52 88 46 68 88 a4 6b  |=.@......R.Fh..k|
     00000080  1f 88 ac 68 68                                    |...hh|
    }
   },
   Sequence: (uint32) 0
  })
 },
 TxOut: ([]*wire.TxOut) (len=1 cap=1) {
  (*wire.TxOut)(0xc0017d2280)({
   Value: (int64) 4210,
   PkScript: ([]uint8) (len=34 cap=34) {
    00000000  00 20 65 57 9d 0f bc 9d  53 63 c1 04 38 95 a8 be  |. eW....Sc..8...|
    00000010  ae 32 4a 55 54 f3 e6 75  55 de d1 f7 67 9f f5 6c  |.2JUT..uU...g..l|
    00000020  fb 9f                                             |..|
   }
  })
 },
 LockTime: (uint32) 1352192
})

2019-03-15 11:39:41.480 [ERR] UTXN: Unable to broadcast baby tx: Transaction rejected: output already spent, (*wire.MsgTx)(0xc001a7a200)({
 Version: (int32) 2,
 TxIn: ([]*wire.TxIn) (len=1 cap=1) {
  (*wire.TxIn)(0xc0000b46c0)({
   PreviousOutPoint: (wire.OutPoint) 59562377bafd4de8fe5ecad7ae69c18a64b3119d041868552869cf5048bc7336:0,
   SignatureScript: ([]uint8) {
   },
   Witness: (wire.TxWitness) (len=5 cap=5) {
    ([]uint8) {
    },
    ([]uint8) (len=71 cap=71) {
     00000000  30 44 02 20 54 6c 18 59  85 59 6d 43 84 3b bf 6b  |0D. Tl.Y.YmC.;.k|
     00000010  b0 ee 55 cd 0e f8 2f 25  72 c7 b9 6e 72 29 38 2b  |..U.../%r..nr)8+|
     00000020  fa 6d f1 45 02 20 54 d7  ce dd c4 1b e3 0a a4 b1  |.m.E. T.........|
     00000030  49 24 05 d1 a3 98 9b ac  dd b3 f1 19 f5 e9 25 f8  |I$............%.|
     00000040  c4 59 53 44 b1 1c 01                              |.YSD...|
    },
    ([]uint8) (len=71 cap=71) {
     00000000  30 44 02 20 50 6b 7c 99  97 34 c3 5b 6e e7 7d 61  |0D. Pk|..4.[n.}a|
     00000010  05 44 38 4e 04 be e7 b4  56 47 48 91 02 b5 f2 8d  |.D8N....VGH.....|
     00000020  6a 55 a7 a7 02 20 46 5c  c0 f4 7c 17 c8 1b f1 a8  |jU... F\..|.....|
     00000030  6e df b6 6d 85 45 ef 4c  8e 0d 80 39 c5 e6 e3 5d  |n..m.E.L...9...]|
     00000040  bd e8 21 d5 38 cd 01                              |..!.8..|
    },
    ([]uint8) {
    },
    ([]uint8) (len=133 cap=133) {
     00000000  76 a9 14 59 43 52 37 86  fc 62 17 6d 84 ba 57 3c  |v..YCR7..b.m..W<|
     00000010  ff 14 52 56 57 a7 cd 87  63 ac 67 21 02 2a 4c cc  |..RVW...c.g!.*L.|
     00000020  f1 5f 8d 78 d3 91 79 5c  d7 28 6d 29 79 1a d8 8f  |._.x..y\.(m)y...|
     00000030  79 9c 49 b1 36 a3 00 2f  24 8b 37 4a af 7c 82 01  |y.I.6../$.7J.|..|
     00000040  20 87 64 75 52 7c 21 02  3d f2 bd 25 63 00 65 0e  | .duR|!.=..%c.e.|
     00000050  08 6b 17 e9 7c 0f 81 c5  01 c9 03 02 47 79 bb d9  |.k..|.......Gy..|
     00000060  6f 5b b3 e8 8d c5 80 2c  52 ae 67 a9 14 e2 57 b9  |o[.....,R.g...W.|
     00000070  3d a8 40 1a a9 d6 87 7f  9b 52 88 46 68 88 a4 6b  |=.@......R.Fh..k|
     00000080  1f 88 ac 68 68                                    |...hh|
    }
   },
   Sequence: (uint32) 0
  })
 },
 TxOut: ([]*wire.TxOut) (len=1 cap=1) {
  (*wire.TxOut)(0xc0017d2280)({
   Value: (int64) 4210,
   PkScript: ([]uint8) (len=34 cap=34) {
    00000000  00 20 65 57 9d 0f bc 9d  53 63 c1 04 38 95 a8 be  |. eW....Sc..8...|
    00000010  ae 32 4a 55 54 f3 e6 75  55 de d1 f7 67 9f f5 6c  |.2JUT..uU...g..l|
    00000020  fb 9f                                             |..|
   }
  })
 },
 LockTime: (uint32) 1352192
})

2019-03-15 11:39:41.480 [ERR] UTXN: Failed to sweep first-stage HTLC (CLTV-delayed) output 3c92ea7e140bedb36bad976d807dc2f33b41d5e1412a6176708b7756b3b2d06e:0
2019-03-15 11:39:41.482 [ERR] UTXN: Failed to sweep outputs at height=1352192: Transaction rejected: output already spent
2019-03-15 11:39:41.482 [ERR] SRVR: unable to start server: Transaction rejected: output already spent

2019-03-15 11:39:41.482 [INF] RPCS: Stopping RPC Server
2019-03-15 11:39:41.486 [INF] LTND: Shutdown complete
```

**Comment by Roasbeef** [comment]
Will be fixed in #2800.

**Comment by Roasbeef** [comment]
Closing as duplicate of #2793.
